Fix incorrect context length for llama3.2-11b

rchen19 opened the pull request:
## Motivation

In `python/sglang/srt/managers/tokenizer_manager.py` file, the `__init__` method of class `TokenizerManager`, this line:

```python
self.context_len = server_args.context_length or get_context_length(
            self.hf_config
        )
```

is problematic because when loading a vision-LLM like Llama-3.2, the text related config is under `config/text_config`, and here `self.hf_config` is the whole model config, not just `text_config`, `get_context_length()` function won’t find max context length parameter and will instead return a default value of 2048. 

## Modifications

A simple fix is just replacing this line with:

```python
self.context_len = server_args.context_length or get_context_length(
            getattr(self.hf_config, "text_config", self.hf_config)
        )
```

to check if there is a `text_config` item inside the whole config, if yes, there the context length parameter must be found under that.

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [ ] (not sure if this is needed for such a small change, as there is not a general test for making sure the model config is read correctly from HF repos) Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [ ] (not needed) Update documentation as needed, including docstrings or example tutorials.

rchen19 committed:
correcly read context length from text_config of the model is multimodal

rchen19 committed:
Merge branch 'sgl-project:main' into fix-incorrect-context-length-for-llama32

rchen19 committed:
Merge branch 'sgl-project:main' into fix-incorrect-context-length-for-llama32

merrymercy reviewed (commented):
There are multiple places that call `get_context_length`.
Can you fix do the fix inside of `get_context_length`?

rchen19 committed:
fix in `get_context_length` instead of `TokenizerManager`

- noticing the original `get_context_length` is also NOT getting `rope_scaling` from the text config correctly

rchen19 committed:
Merge remote-tracking branch 'refs/remotes/origin/fix-incorrect-context-length-for-llama32' into fix-incorrect-context-length-for-llama32

rchen19 committed:
Merge branch 'sgl-project:main' into fix-incorrect-context-length-for-llama32

rchen19 commented:
> There are multiple places that call `get_context_length`. Can you fix do the fix inside of `get_context_length`?

Done.

And you are right, the rope_scaling factor was also not correctly read from the text_config.
